spec/controllers/books_controller_spec.rb

```diff
@@ -0,0 +1,89 @@
+require 'rails_helper'
+
+describe BooksController do
+  let!(:books) { create_list(:book, 3) }
+
+  describe 'GET #index' do
+    context 'With an unauthenticated user' do
+      it 'returns an unauthorized status' do
+        get :index, params: {}
+        expect(response).to have_http_status(:unauthorized)
+      end
+    end
+
+    context 'With an authenticated user' do
+      include_context 'Authenticated User'
+
+      it 'without params returns all books' do
+        get :index
+        expect(JSON.parse(response.body)['total_count']).to eq(books.size)
+      end
+
+      it 'returns a 200 ok status' do
+        get :index, params: { book: { genre: books.first.genre } }
+        expect(response).to have_http_status(:ok)
+      end
+
+      def unique_title
+        rnd_title = Faker::Book.title
+        rnd_title = Faker::Book.title while books.map(&:title).include?(rnd_title)
+        rnd_title
+      end
+
+      it 'params filters not exists' do
+        get :index, params: { book: { title: unique_title } }
+        expect(JSON.parse(response.body)['total_count']).to eq(0)
+      end
+
+      def full_index_response
+        b = books.first
+        get :index, params: { book: {
+          title: b.title,
+          # Below are additional params
+          image: b.image,
+          publisher: b.publisher,
+          year: b.year
+        } }
+      end
+
+      it 'setting additionals params' do
+        full_index_response
+        expect(response).to have_http_status(:ok)
```

Comment:
What if our filter function is broken? The endpoint will return a 200 but the response will be empty!
I would go a little forward an check that the response contains the desired book.